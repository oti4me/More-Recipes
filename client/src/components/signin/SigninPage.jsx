import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import SigninForm from './SigninForm.jsx'
import Footer from '../Footer.jsx'
import Header from '../Header.jsx'


class SigninPage extends React.Component {

  constructor(props) {
    super(props);
    this.handleSignin.bind(this);
  }

  handleSignin(user) {
    console.log("top level compoment handke signin");
    // this.props.handleSignin(user);
  }

  render() {
    return (
      <div>
        <Header { ...this.props } />
        <div className="container">
          <div className="row">
            <div className="col s12  s2 m4 l4 offset-m4 offset-l4">
              <h2 className="top-margin-50">Sign In</h2>
              <SigninForm handleSignin={this.handleSignin} history={this.props.history} />
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { state };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(SigninPage);
