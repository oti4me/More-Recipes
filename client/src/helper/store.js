import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducers from '../reducers';

// initial state of the app
const initialState = {
  recipes: {},
  auth: {
    user: {},
    loggedIn: false,
  }
};

// create store to save app state
const store = createStore(reducers, initialState,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;