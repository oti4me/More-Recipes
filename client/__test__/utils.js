import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import store from '../src/helper/createStore';

export const mountWrapper = (Component, store, props) => mount(
  <Provider store={store} >
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  </Provider>
);

export const mainState = store.getState();

export default {
  mountWrapper,
}