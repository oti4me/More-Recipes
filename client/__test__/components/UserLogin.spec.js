import React from 'react';
import toJson from 'enzyme-to-json';
// import ConnectedSignupForm, { SignupForm } from '../../src/components/signup/SignupForm';
// import ConnectedSignupPage, { SignupPage } from '../../src/components/signup/SignupPage';
import SigninPage from '../../src/components/signin/SigninPage';
import { mountWrapper } from '../utils';
import mockStore from '../mockData/mockStore';
import store from '../../src/helper/createStore';

const mainState = store.getState();
const props = {
  history: {
    push: () => { },
  }
};

const newState = {
  ...mainState,
  getFavourites: jest.fn(() => Promise.resolve()),
  auth: {
    user: {
      userId: 3,
    },
    loggedIn: true,
  }
};

const event = (name, value) => ({
  preventDefault: jest.fn(),
  target: {
    name,
    value
  }
});

const userInfo = {
  email: 'otighe@gmail.com',
  password: 'Oti4me@gmail.com'
}

const newStore = mockStore(newState);
const wrapper = mountWrapper(SigninPage, newStore, props);
const SigninFormFormWrapper = wrapper.find('SigninForm');

describe('Sign-Up Component Suite', () => {
  it('should render SignupPage correctly without exploding', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render SignupPage correctly without exploding', () => {
    SigninFormFormWrapper.instance()
      .handleChange(event('email', userInfo.email));
    SigninFormFormWrapper.instance()
      .handleChange(event('password', userInfo.password));
  });

  it('should render SignupPage correctly without exploding', () => {
    SigninFormFormWrapper.find('a').at(0).simulate('click', {
      preventDefault: jest.fn(),
    });
  });

  // it('should hadle buttons', () => {
  //   SignupFormWrapper.instance()
  //     .handleChange({
  //       name: 'firstName',
  //       value: 'Henry'
  //     }, event);

  //   SignupFormWrapper.instance.setState(userInfo);
  //   expect(favouriteListWrapper.instance().state)
  //     .toEqual(userInfo);
  //   favouriteListWrapper.find('a').at(0).simulate('click', {
  //     preventDefault: jest.fn(),
  //   });
  //   favouriteListWrapper.find('a').at(1).simulate('click', {
  //     preventDefault: jest.fn(),
  //   });
  // });
});


// import React from 'react';
// import SigninForm2, { SigninForm } from '../../src/components/signin/SigninForm';
// import SigninPage from '../../src/components/signin/SigninPage';


// const state = {
//   email: "",
//   password: ""
// };

// const func = jest.fn();
// const props = {
//   loggedIn: true,
//   error: {},
//   history: {},
//   signin: jest.fn()
// };
// const signin = jest.fn()

// const errorProps = {
//   loggedIn: true,
//   error: {
//     status: 400,
//     message: [
//       'username cannot be empty'
//     ]
//   },
//   signin: jest.fn()
// };

// const event = {
//   target: {
//     name: 'firstName',
//     value: 'Henry'
//   },
//   preventDefault: jest.fn()
// };
// describe('Should handle change', () => {

//   it('Should render component without exploding', () => {
//     const wrapper = shallow(<SigninForm {...props} />);
//     expect(wrapper).toMatchSnapshot();
//   });

//   it('Should render component without exploding', () => {
//     const wrapper = shallow(<SigninForm2 store={store} {...props} />);
//     expect(wrapper.length).toBe(1);
//   });


//   it('Should handle field change', () => {
//     const wrapper = shallow(<SigninForm signin={signin} {...props} {...state} />);
//     wrapper.instance().handleChange(event);
//     wrapper.instance().handleSignin(event);
//     wrapper.find('#form').simulate('submit', event);
//     expect(props.signin.mock.calls.length).toBe(1)
//   });
//   it('Should handle signin button clicked', () => {
//     const wrapper = shallow(<SigninForm signin={signin} {...props} {...state} />);
//     wrapper.find('#signin').simulate('click', event);
//     expect(props.signin.mock.calls.length).toBe(2);
//   });
// });

// describe('Signin page suit', () => {
//   it('Should render signin page without exploding', () => {
//     const wrapper = shallow(<SigninPage {...props} />);
//     expect(wrapper).toMatchSnapshot();
//   });
// });