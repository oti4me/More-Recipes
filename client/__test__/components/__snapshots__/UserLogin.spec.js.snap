// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-Up Component Suite should render SignupPage correctly without exploding 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SigninPage
        history={
          Object {
            "push": [Function],
          }
        }
      >
        <div
          className="main"
        >
          <Connect(Header)
            history={
              Object {
                "push": [Function],
              }
            }
          >
            <Header
              history={
                Object {
                  "push": [Function],
                }
              }
              loggedIn={true}
              signOut={[Function]}
              user={
                Object {
                  "userId": 3,
                }
              }
            >
              <div>
                <nav>
                  <div
                    className="nav-wrapper color-green header1"
                  >
                    <Link
                      className="brand-logo"
                      replace={false}
                      style={
                        Object {
                          "fontSize": "22px",
                        }
                      }
                      to="/home"
                    >
                      <a
                        className="brand-logo"
                        href="/home"
                        onClick={[Function]}
                        style={
                          Object {
                            "fontSize": "22px",
                          }
                        }
                      >
                        More-Recipes
                      </a>
                    </Link>
                    <Link
                      className="button-collapse"
                      data-activates="mobile2"
                      replace={false}
                      to="!#"
                    >
                      <a
                        className="button-collapse"
                        data-activates="mobile2"
                        href="!#"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons"
                        >
                          menu
                        </i>
                      </a>
                    </Link>
                    <ul
                      className="right hide-on-med-and-down"
                      id="nav-mobile"
                      style={
                        Object {
                          "width": "330px",
                        }
                      }
                    >
                      <li>
                        <Link
                          replace={false}
                          to="/home"
                        >
                          <a
                            href="/home"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-home"
                            >
                               Home
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li>
                        <Link
                          className="dropdown-button"
                          data-activates="dropdown2"
                          replace={false}
                          to="!#"
                        >
                          <a
                            className="dropdown-button"
                            data-activates="dropdown2"
                            href="!#"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-user-circle-o"
                            >
                               
                              User Account
                               
                              <i
                                aria-hidden="true"
                                className="fa fa-angle-down"
                              />
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li>
                        <Link
                          onClick={[Function]}
                          replace={false}
                          to="!#"
                        >
                          <a
                            href="!#"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-sign-out"
                            >
                               Logout
                            </i>
                          </a>
                        </Link>
                      </li>
                      <ul
                        className="dropdown-content"
                        id="dropdown2"
                        style={
                          Object {
                            "marginTop": "50px",
                          }
                        }
                      >
                        <li>
                          <Link
                            replace={false}
                            to="/profile"
                          >
                            <a
                              href="/profile"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-user-circle"
                              >
                                 
                                 Profile
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/addrecipe"
                          >
                            <a
                              href="/addrecipe"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-plus-square"
                              >
                                 
                                 Add Recipe
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/myrecipes"
                          >
                            <a
                              href="/myrecipes"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-cutlery"
                              >
                                 
                                 My Recipes
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/favourites"
                          >
                            <a
                              href="/favourites"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-heart"
                              >
                                 
                                 Favourites
                              </i>
                            </a>
                          </Link>
                        </li>
                      </ul>
                    </ul>
                    <ul
                      className="side-nav"
                      id="mobile2"
                    >
                      <li>
                        <Link
                          replace={false}
                          to="/home"
                        >
                          <a
                            href="/home"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-home"
                            >
                              Home
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li
                        className="divider"
                      />
                      <li>
                        <Link
                          replace={false}
                          to="/profile"
                        >
                          <a
                            href="/profile"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-user-circle"
                            >
                              Profile
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li
                        className="divider"
                      />
                      <li>
                        <Link
                          replace={false}
                          to="/addrecipe"
                        >
                          <a
                            href="/addrecipe"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-plus-square"
                            >
                              Add Recipe
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li
                        className="divider"
                      />
                      <li>
                        <Link
                          replace={false}
                          to="/myrecipes"
                        >
                          <a
                            href="/myrecipes"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-cutlery"
                            >
                              My Recipes
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li
                        className="divider"
                      />
                      <li>
                        <Link
                          replace={false}
                          to="/favourites"
                        >
                          <a
                            href="/favourites"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-heart"
                            >
                              Favourites
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li
                        className="divider"
                      />
                      <li>
                        <Link
                          id="signOut"
                          onClick={[Function]}
                          replace={false}
                          to="!#"
                        >
                          <a
                            href="!#"
                            id="signOut"
                            onClick={[Function]}
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-sign-out"
                            >
                               Logout
                            </i>
                          </a>
                        </Link>
                      </li>
                      <li
                        className="divider"
                      />
                    </ul>
                  </div>
                </nav>
              </div>
            </Header>
          </Connect(Header)>
          <div
            className="container cont"
          >
            <div
              className="row"
            >
              <div
                className="col s12  s2 m6 l6 offset-m3 offset-l3"
              >
                <h2
                  className="top-margin-50"
                >
                  Sign In
                </h2>
                <Connect(SigninForm)
                  history={
                    Object {
                      "push": [Function],
                    }
                  }
                >
                  <SigninForm
                    history={
                      Object {
                        "push": [Function],
                      }
                    }
                    loggedIn={true}
                    signin={[Function]}
                  >
                    <div
                      className="row"
                    >
                      <form
                        className="col s12 z-depth-2"
                        id="form"
                        style={
                          Object {
                            "padding": "50px",
                          }
                        }
                      >
                        <div
                          className="row"
                        >
                          <div
                            className="input-field col s12"
                          >
                            <i
                              className="material-icons prefix"
                            >
                              email
                            </i>
                            <input
                              id="email"
                              name="email"
                              onChange={[Function]}
                              type="text"
                              value=""
                            />
                            <label
                              htmlFor="email"
                            >
                              Email
                            </label>
                          </div>
                        </div>
                        <div
                          className="row"
                        >
                          <div
                            className="input-field col s12"
                          >
                            <i
                              className="material-icons prefix"
                            >
                              vpn_key
                            </i>
                            <input
                              id="password"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value=""
                            />
                            <label
                              htmlFor="password"
                            >
                              Password
                            </label>
                          </div>
                        </div>
                        <div
                          className="row"
                        >
                          <Link
                            className="waves-effect waves-light btn"
                            id="signin"
                            onClick={[Function]}
                            replace={false}
                            role="button"
                            tabIndex="0"
                            to="#!"
                          >
                            <a
                              className="waves-effect waves-light btn"
                              href="#!"
                              id="signin"
                              onClick={[Function]}
                              role="button"
                              tabIndex="0"
                            >
                              Sign In
                            </a>
                          </Link>
                        </div>
                        <div
                          className="row"
                        >
                          <p
                            className=""
                          >
                            Dont have an account?
                            <Link
                              replace={false}
                              to="/signup"
                            >
                              <a
                                href="/signup"
                                onClick={[Function]}
                              >
                                 Sign Up
                              </a>
                            </Link>
                          </p>
                        </div>
                      </form>
                    </div>
                  </SigninForm>
                </Connect(SigninForm)>
              </div>
            </div>
          </div>
          <Footer>
            <footer
              className="foot"
            >
              <p
                className="foo"
              >
                Copyright © 2017, More Recipes, All rights reserved
              </p>
            </footer>
          </Footer>
        </div>
      </SigninPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
